### ISSUE TITLE
Failing parallel builds using MSBuild

### ISSUE THREAD

**Issue by MathiasMagnus** [KhronosGroup/OpenCL-CLHPP#162]
When using MSBuild (aka. the Visual Studio generators of CMake) when trying to build in parallel, multiple invocations are needed to complete a build. Only the Ninja and Makefile generators seem to follow dependencies properly.

```
cmake.exe -G "Visual Studio 17 2022" -A x64 ... -S ./OpenCL-CLHPP -B ./OpenCL-CLHPP/build
cmake.exe --build ./OpenCL-CLHPP/build --target install -- /m /v:m
```

Has failures.

**Comment by bashbaug** [comment]
I think this is a duplicate of #75.

Are we missing dependency information in our CMake file or is there something fundamentally broken here?

**Comment by MathiasMagnus** [comment]
Indeed, it's a duplicate. I fixed this in our branch too, so fix incoming. The problem is that MSBuild seems to only track outputs at a target level and not at the custom command level, therefore multiple targets in parallel invoking the very same custom commands collide and some fail because they are trying to write the file with the same name from multiple processes.

Elevating the custom commands to targets solves this issue. The linked execute process solution also works, as it goes around collision by moving the problematic part to configuration time.

Targets work better for working with the tests and reconfigures faster, whereas the config-time solution "pollutes" less the list of targets by not introducing new ones.

**Comment by bashbaug** [comment]
Closing - fixed!
